### PR TITLE
Add a contributing guideline

### LINKED ISSUE #46: Add a contributing guideline
## Why

To help the other developer can easy to run it locally or contribute. It's better to add some content like 
- How to install, and run the codebase on local
- How to build and run the app on local
- Explain the structure of this template

The is a file named [`contributing.md`](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors) that is supported by GitHub for this purpose

## Who Benefits?

Contributors

### PR DESCRIPTION
**Pull request by hoangmirs**
- Close #46

## What happened 👀

Add a contributing guideline to involve people contributing to the project

## Insight 📝

Write the content to `CONTRIBUTING.md` file following the [Github documentation](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors#adding-a-contributing-file)

## Proof Of Work 📹

N/A

### CODE REVIEW

**Review by byhbt — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+## Issues and feature requests
+
+If you have any issues or feature requests, please create an issue in the [GitHub issues](https://github.com/nimblehq/infrastructure-templates/issues) page.
+
+## Development
+
+### Prerequisites
+
+- Node.js 12.0.0 or higher
+- NPM is preferred to use with this project, but you can use Yarn as well and don't commit the `yarn.lock` file to the repository
```
Comment (andyduong1920) [inline CONTRIBUTING.md:20]:
```suggestion
- NPM is preferred to use with this project
```

Let remove the Yarn topic to make the sentence less confuse :D

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+npm run lint // to check linting
+
+npm run lint:fix // to fix linting
+```
+
+## Publishing
+
+- This project will be published to NPM automatically when a new release is created in GitHub. Therefore, the package version in `package.json` should be updated before creating a new release.
+
+- The release should be created in the `master` branch.
```
Comment (andyduong1920) [inline CONTRIBUTING.md:96]:
```suggestion
- The release should be created in the `main` branch.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
+# Contributing guidelines for the project
+
+First off, thank you for considering contributing to this project. It's people like you that make this project such a great tool. We appreciate your help!
+
+The following is a set of guidelines for contributing to this project. These are mostly guidelines, not rules. Use your best judgment, and feel free to propose changes to this document in a pull request.
+
+## Introduction
+
+This project is using TypeScript and [OCLIF](https://oclif.io/) to build the CLI. The project is using Jest for testing, ESLint for code formatting and linting.
```
Comment (nvminhtue) [inline CONTRIBUTING.md:9]:
```suggestion
This project uses TypeScript and [OCLIF](https://oclif.io/) to build the CLI, Jest for testing, and ESLint for code formatting and linting.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    - `core` - contains the code to generate the main Terraform files e.g. `main.tf`, `variables.tf`, etc.
+  - `skeleton` - contains the skeleton files for the project
+    - `addons` - contains the skeleton files for common addons e.g. version control, CI/CD, etc.
+    - `aws` - contains the skeleton files for AWS modules
+    - `core` - contains the skeleton files for common files e.g. `main.tf`, `variables.tf`, `.gitignore`, etc.
+  - `test` - contains the test helpers and configurations
+
+### Add a new command
+
+To add a new command, you need to create a new folder in the `src/commands` folder and add the `index.ts` file in it.
```
Comment (nvminhtue) [inline CONTRIBUTING.md:63]: maybe we should say something like this
```suggestion
To add a new command, create a new folder in the `src/commands` folder and add the `index.ts` file.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+### Manual publishing
+
+- To publish the project manually, run the following command:
+
+```bash
+npm run publish
+```
+
+**Note:** You need to have the NPM account to publish the project and make sure that the version in `package.json` is updated.
```
Comment (nvminhtue) [inline CONTRIBUTING.md:108]:
```suggestion
**Note:** NPM account is required to publish the project. Ensure that the version in `package.json` is updated.
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (hoangmirs) [reply CONTRIBUTING.md]: Modified in https://github.com/nimblehq/infrastructure-templates/pull/141/commits/1b6304955c0ca8410a0a48d67a6894b2b425383e

File: CONTRIBUTING.md (hunk 2 above)
Comment (hoangmirs) [reply CONTRIBUTING.md]: Modified in https://github.com/nimblehq/infrastructure-templates/pull/141/commits/1b6304955c0ca8410a0a48d67a6894b2b425383e

File: CONTRIBUTING.md (hunk 3 above)
Comment (hoangmirs) [reply CONTRIBUTING.md]: Modified in https://github.com/nimblehq/infrastructure-templates/pull/141/commits/1b6304955c0ca8410a0a48d67a6894b2b425383e

File: CONTRIBUTING.md (hunk 4 above)
Comment (hoangmirs) [reply CONTRIBUTING.md]: Modified in https://github.com/nimblehq/infrastructure-templates/pull/141/commits/1b6304955c0ca8410a0a48d67a6894b2b425383e

File: CONTRIBUTING.md (hunk 5 above)
Comment (hoangmirs) [reply CONTRIBUTING.md]: Modified in https://github.com/nimblehq/infrastructure-templates/pull/141/commits/1b6304955c0ca8410a0a48d67a6894b2b425383e

**Review by andyduong1920 — APPROVED** [review]

**Review by nvminhtue — APPROVED** [review]

**Pull request merged**